kafka-handler/src/java/org/apache/hadoop/hive/kafka/KafkaSerDe.java

```diff
@@ -133,12 +134,40 @@
       Preconditions.checkArgument(!schemaFromProperty.isEmpty(), "Avro Schema is empty Can not go further");
       Schema schema = AvroSerdeUtils.getSchemaFor(schemaFromProperty);
       LOG.debug("Building Avro Reader with schema {}", schemaFromProperty);
-      bytesConverter = new AvroBytesConverter(schema);
+      bytesConverter = getByteConverterForAvroDelegate(schema, tbl);
     } else {
       bytesConverter = new BytesWritableConverter();
     }
   }
 
+  enum BytesConverterType {
+    CONFLUENT,
+    SKIP,
+    NONE;
+
+    static BytesConverterType fromString(String value) {
+      try {
+        return BytesConverterType.valueOf(value.trim().toUpperCase());
+      } catch (Exception e){
+        return NONE;
+      }
+    }
+  }
+
+  BytesConverter getByteConverterForAvroDelegate(Schema schema, Properties tbl) {
+    String avroBytesConverterProperty = tbl.getProperty(AvroSerdeUtils
+                                                            .AvroTableProperties.AVRO_SERDE_TYPE
+                                                            .getPropName(), BytesConverterType.NONE.toString());
+    BytesConverterType avroByteConverterType = BytesConverterType.fromString(avroBytesConverterProperty);
+    Integer avroSkipBytes = Integer.getInteger(tbl.getProperty(AvroSerdeUtils.AvroTableProperties.AVRO_SERDE_SKIP_BYTES
```

Comment:
not sure how to read this ? looking at the implementation of `Integer.getInteger` i see it is reading form the system properties, this is clearly not going to work. Aha... Java naming

Reply:
` public static Integer getInteger(String nm, Integer val) {
        String v = null;
        try {
            v = System.getProperty(nm);
        } catch (IllegalArgumentException | NullPointerException e) {
        }
        if (v != null) {
            try {
                return Integer.decode(v);
            } catch (NumberFormatException e) {
            }
        }
        return val;
    }`

Reply:
Yeah, we are basically getting the property name for the number of bytes to read, then getting the string value, then converting that to an integer. I think it might be easier to read with an intermediate variable. I'll add that.

Reply:
Fixed. Please check that the new code is more readable and resolve this issue.

Reply:
@davidov541 did you read the comment above ? That function you are using is reading from the system property of the JVM please see the function implementation `System.getProperty(nm)`.

Reply:
I think I'm confused what you're asking for then. The initialize function takes in a java.util.Properties object that has properties that have been set for the serde in the DDL for the table. It reads a few from that object, and then passes it to getByteConverterForAvroDelegate, where it is also used in the code added here. The usage of the properties object here matches what is being done in initialize, and seems to match what I would expect. These aren't pulling system properties of the JVM, or at least are not necessarily doing so, instead reading from the Properties object passed to us.

Does that make sense, or am I way off base?

Reply:
@davidov541 please read the java doc of `java.lang.Integer#getInteger(java.lang.String)` In the first line it says
`* Determines the integer value of the system property with the
     * specified name.`
Reading the code should help as well ....

Reply:
tl;dr - Just use `Integer.parseInt(tbl.getProperty())`

Reply:
Thanks @b-slim , I definitely skipped that multiple times when I was reading the code. That is a bizarre issue, but I'm glad you pointed it out. I've fixed it, and am running the tests now.

Reply:
FYI the goal of code review is not to tell devs what to do, but to help them understand why the code need to be changed ....

Reply:
OK, made the fix. Please check it to make sure I got it right this time.